src/js/components/ArticleForm.js

```diff
@@ -12,19 +12,37 @@ export default React.createClass({
     return {title: "", url: ""};
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    this.setState({
+      title: nextProps.current && nextProps.current.title || "",
+      url:   nextProps.current && nextProps.current.url || ""
+    });
+  },
+
   handleSubmit: function(event) {
     event.preventDefault();
-    ArticleActions.create({
+    var params = Object.assign({
+      id: this.props.current && this.props.current.id
```

Comment:
When I read this idiom I expect a boolean :(  (*convention or seek for brevity?*)